### PR TITLE
Renamed ExtractionError to ExtractionWarning #2378

### PR DESCRIPTION
**Pull request by Onager**
Rationale for this change is in #2378.

Requests/Notes for reviewer:
- Much of this change is a simple rename
- The most interesting parts of this change are probably in plaso/storage/sqlite/sqlite_file.py and include backwards-compatible functionality.

### CODE REVIEW

File: plaso/storage/sqlite/merge_reader.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import sqlite3
 import zlib
 
-from plaso.containers import errors
+from plaso.containers import warnings
```
Comment (joachimmetz) [inline plaso/storage/sqlite/merge_reader.py]: please use alphabetical order

File: plaso/storage/sqlite/sqlite_file.py [hunk 2]
```diff
@@ -237,7 +239,7 @@ def _CountStoredAttributeContainers(self, container_type):
       raise ValueError('Attribute container type {0:s} is not supported'.format(
           container_type))
 
-    if not self._HasTable(self._CONTAINER_TYPE_ANALYSIS_REPORT):
```
Comment (joachimmetz) [inline plaso/storage/sqlite/sqlite_file.py]: Maybe fix this in a separate PR and add test coverage.

File: plaso/storage/sqlite/sqlite_file.py [hunk 3]
```diff
@@ -719,13 +722,35 @@ def GetAnalysisReports(self):
...
+    """Retrieves the warnings.
 
     Returns:
-      generator(ExtractionError): error generator.
+      generator(ExtractionWarning): warning generator.
+    """
+    # For backwards compatibility with pre-20190309 stores.
+    # Note that stores cannot contain both ExtractionErrors and
+    # ExtractionWarnings
+    if self._HasAttributeContainers(self._CONTAINER_TYPE_EXTRACTION_ERROR):
```
Comment (joachimmetz) [inline plaso/storage/sqlite/sqlite_file.py]: What if I open an existing store and add more events and warnings?

File: plaso/storage/sqlite/sqlite_file.py (hunk 2 above)
Comment (Onager) [reply plaso/storage/sqlite/sqlite_file.py]: Done - #2402

File: plaso/storage/sqlite/merge_reader.py (hunk 1 above)
Comment (Onager) [reply plaso/storage/sqlite/merge_reader.py]: Done

File: plaso/storage/sqlite/sqlite_file.py (hunk 3 above)
Comment (Onager) [reply plaso/storage/sqlite/sqlite_file.py]: Done in #2407

**Comment by joachimmetz** [comment]
Waiting for tests to pass

**Comment by joachimmetz** [comment]
@Onager tests seems to be failing.

**Pull request merged**